src/index.js

```diff
@@ -1,39 +1,3 @@
-const semver = require('semver')
-const core = require('@actions/core')
-const github = require('./github')
-const utils = require('./utils')
+const run = require('./run').run
 
-const run = async () => {
-  const octokit = github.getOctokit(core.getInput('github-token'))
-
-  const [owner, repo] = process.env.GITHUB_REPOSITORY.split('/')
-  const sha = process.env.GITHUB_SHA
-
-  let latestTag
-  try {
-    latestTag = await github.getLatestTag(octokit, owner, repo)
-  } catch (err) {
-    return core.setFailed(`unable to get latest tag - error: ${err.message} ${err?.response?.status}`)
-  }
-
-  // return a default version if no previous github tags
-  if (!latestTag) {
-    const incrementedVersion = semver.inc('0.0.0', core.getInput('default-bump'))
-    return utils.setVersionOutputs(incrementedVersion, core.getInput('prefix'))
-  }
-
-  if (!semver.valid(latestTag.name)) {
-    return core.setFailed(`latest tag name is not valid semver: ${JSON.stringify(latestTag)}`)
-  }
-
-  // get commits from last tag and calculate version bump
-  const commits = await github.compareCommits(octokit, owner, repo, latestTag?.commit?.sha, sha)
-  const bump = await utils.getVersionBump(commits, core.getInput('default-bump'))
-
-  const incrementedVersion = semver.inc(latestTag.name, bump)
-  utils.setVersionOutputs(incrementedVersion, core.getInput('prefix'))
-}
-
-if (process.env.NODE_ENV !== 'test') run()
-
-module.exports = { run }
+run()
```

Comment:
Do you think it provides any value to wrap this within a try/catch to provide a cleaner error message for any uncaught exceptions? Or maybe we just let the application fail and show the full Node error message?

Something like this - although I'm not 100% sure if that `run()` needs to have an `await` before it. It may not matter since there is no other code after it.

```js
const core = require('@actions/core')
const run = require('./run').run

try {
  run() 
} catch (err) {
  core.setFailed(err.message)
}
```

Reply:
Possible patch for this https://github.com/mdb/action-conventional-versioning/pull/1/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R1-R8

Reply:
@jveldboom Good call! Although, should `run()` be `await run()`?